fastfold/distributed/comm.py

```diff
@@ -27,31 +27,48 @@ def _reduce(tensor: Tensor) -> Tensor:
     return tensor
 
 
-def _split(tensor: Tensor, dim: int = -1) -> Tensor:
+def _split(tensor: Tensor, dim: int = -1, drop_unused=False) -> Tensor:
     if gpc.get_world_size(ParallelMode.TENSOR) == 1:
         return tensor
 
     split_size = divide(tensor.shape[dim], gpc.get_world_size(ParallelMode.TENSOR))
     tensor_list = torch.split(tensor, split_size, dim=dim)
+    rank = gpc.get_local_rank(ParallelMode.TENSOR)
 
-    output = tensor_list[gpc.get_local_rank(ParallelMode.TENSOR)].contiguous()
+    if not drop_unused:
+        output = tensor_list[rank].contiguous()
+    else:
+        output = tensor_list[rank].contiguous().clone()
```

Comment:
Hi, thanks for your contribution.
But I am not sure of the meaning of `drop_unused` and this line. In my understanding, this line only creates a totally same tensor of `tensor_list[rank]`.

Reply:
Yes, but it needs to be used with  torch.cuda.empty_cache() to be effective.

Reply:
I have tried your code. Here is the question:
The`torch.cuda.max_memory_allocated()` increased, which means it makes the peak memory increasing.
I think your motivation idea is: with `.clone()` and `torch.cuda.empty_cache()` to escape `tensor_list`, only keep `tensor_list[rank]` to save memory. However, when `.clone()`, there are both `tensor_list` and `output` in the memory, leading to an increase of peak memory.